it/connectors/DepartureMessageConnectorSpec.scala

```diff
@@ -0,0 +1,204 @@
+      val connector = app.injector.instanceOf[DepartureMessageConnector]
+      val departure = Departure(1, routes.DeparturesController.getDeparture("1").urlWithContext, routes.DepartureMessagesController.getDepartureMessages("1").urlWithContext, Some("MRN"), "ref", "status", LocalDateTime.now, LocalDateTime.now)
+
+      val response = ResponseDeparture(departure)
+      server.stubFor(
+        get(
+          urlEqualTo("/transits-movements-trader-at-departure/movements/departures/1/messages")
+        ).willReturn(aResponse().withStatus(OK)
+          .withBody(Json.toJson(response).toString())))
+
+      implicit val hc = HeaderCarrier()
+      implicit val requestHeader = FakeRequest()
+
+      val result = connector.getMessages("1").futureValue
+
+      result.isLeft mustEqual true
+      result.left.map { x => x.status mustEqual INTERNAL_SERVER_ERROR }
+    }
+
+    "must return HttpResponse with a not found if not found" in {
+      val connector = app.injector.instanceOf[DepartureMessageConnector]
+      server.stubFor(
+        get(
+          urlEqualTo("/transits-movements-trader-at-departure/movements/departures/1/messages")
+        ).willReturn(aResponse().withStatus(NOT_FOUND)))
+
+      implicit val hc = HeaderCarrier()
+      implicit val requestHeader = FakeRequest()
+
+      val result = connector.getMessages("1").futureValue
+
+      result.isLeft mustEqual true
+      result.left.map { x => x.status mustEqual NOT_FOUND }
+    }
+
+    "must return HttpResponse with a bad request if there is a bad request" in {
+      val connector = app.injector.instanceOf[DepartureMessageConnector]
+      server.stubFor(
+        get(
+          urlEqualTo("/transits-movements-trader-at-departure/movements/departures/1/messages")
+        ).willReturn(aResponse().withStatus(BAD_REQUEST)))
+
+      implicit val hc = HeaderCarrier()
+      implicit val requestHeader = FakeRequest()
+
+      val result = connector.getMessages("1").futureValue
+
+      result.isLeft mustEqual true
+      result.left.map { x => x.status mustEqual BAD_REQUEST }
+    }
+
+    "must return HttpResponse with an internal server if there is an internal server error" in {
+      val connector = app.injector.instanceOf[DepartureMessageConnector]
+      server.stubFor(
+        get(
+          urlEqualTo("/transits-movements-trader-at-departure/movements/departures/1/messages")
+        ).willReturn(aResponse().withStatus(INTERNAL_SERVER_ERROR)))
+
+      implicit val hc = HeaderCarrier()
+      implicit val requestHeader = FakeRequest()
+
+      val result = connector.getMessages("1").futureValue
+
+      result.isLeft mustEqual true
+      result.left.map { x => x.status mustEqual INTERNAL_SERVER_ERROR }
+    }
+  }
+
+  //TODO: Refactor this and other spec usages to a common trait
+  override protected def portConfigKey: String = "microservice.services.transits-movements-trader-at-departure.port"
+}
```

Comment:
Can we move this to AppConfig for access, and specify the value in application.conf?